File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/SupplyDelivery.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class SupplyDelivery : DomainResource, IResource
{
    public Base.ResourceReference? Patient { get; set; }
    public Base.ResourceReference? Supplier { get; set; }
    public SupplyDelivery_SuppliedItem? SuppliedItem { get; set; }
    public Base.CodeableConcept? Type { get; set; }
    public string? OccurrenceTiming { get; set; }
    public Base.Period? OccurrencePeriod { get; set; }
    public string? Status { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.ResourceReference[]? BasedOn { get; set; }
    public Base.ResourceReference[]? PartOf { get; set; }
    public string? OccurrenceDateTime { get; set; }
    public Base.ResourceReference[]? Receiver { get; set; }
    public Base.ResourceReference? Destination { get; set; }
    public object? Occurrence    
    {
        get
        {
            if (OccurrenceTiming is not null)
            {
                return OccurrenceTiming;
            }
    
            if (OccurrencePeriod is not null)
            {
                return OccurrencePeriod;
            }
    
            if (OccurrenceDateTime is not null)
            {
                return OccurrenceDateTime;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(string))
            {
                OccurrenceTiming = (string)value;return;
            }
    
            if (value?.GetType() == typeof(Base.Period))
            {
                OccurrencePeriod = (Base.Period)value;return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                OccurrenceDateTime = (string)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }

    public class SupplyDelivery_SuppliedItem : BackboneElement
    {
        public object? Item    
        {
            get
            {
                if (ItemReference is not null)
                {
                    return ItemReference;
                }
        
                if (ItemCodeableConcept is not null)
                {
                    return ItemCodeableConcept;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(Base.ResourceReference))
                {
                    ItemReference = (Base.ResourceReference)value;return;
                }
        
                if (value?.GetType() == typeof(Base.CodeableConcept))
                {
                    ItemCodeableConcept = (Base.CodeableConcept)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public Base.Quantity? Quantity { get; set; }
        public Base.ResourceReference? ItemReference { get; set; }
        public Base.CodeableConcept? ItemCodeableConcept { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/SupplyDelivery.py
from typing import Optional, List
from pydantic import *
from base.Period import Period
from base.CodeableConcept import CodeableConcept
from base.Timing import Timing
from base.Quantity import Quantity
from base.DomainResource import DomainResource
from base.Reference import Reference
from base.Identifier import Identifier
from base.BackboneElement import BackboneElement

class SupplyDelivery_SuppliedItem(BackboneElement):
    quantity: Optional[Quantity] = None
    item_reference: Optional[Reference] = None
    item_codeable_concept: Optional[CodeableConcept] = None

class SupplyDelivery(DomainResource):
    patient: Optional[Reference] = None
    supplier: Optional[Reference] = None
    supplied_item: Optional[SupplyDelivery_SuppliedItem] = None
    type: Optional[CodeableConcept] = None
    occurrence_timing: Optional[Timing] = None
    occurrence_period: Optional[Period] = None
    status: Optional[str] = None
    identifier: Optional[List[Identifier]] = None
    based_on: Optional[List[Reference]] = None
    part_of: Optional[List[Reference]] = None
    occurrence_date_time: Optional[str] = None
    receiver: Optional[List[Reference]] = None
    destination: Optional[Reference] = None